src/lib/y2partitioner/widgets/encrypt_label.rb

```diff
@@ -0,0 +1,91 @@
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "cwm"
+require "y2storage"
+require "y2partitioner/device_graphs"
+
+module Y2Partitioner
+  module Widgets
+    # Label for a {Y2Storage::Luks} device
+    class EncryptLabel < CWM::InputField
+      # Constructor
+      #
+      # @param controller [Actions::Controllers::Encryption]
+      # @param enable [Boolean] whether the widget should be enabled on init
+      def initialize(controller, enable: true)
+        textdomain "storage"
+
+        @controller = controller
+        @enable_on_init = enable
+      end
+
+      # @macro seeAbstractWidget
+      def label
+        _("LUKS &Label (optional)")
```

Comment:
But label is a LUKS2 option, isn't it? See https://github.com/yast/yast-storage-ng/pull/1245/files#diff-0132490c7399997748eabe61f3087f14fa113e28c459f75059da0c3bec1693e2R186-R192

Reply:
Yes, it's the "LUKS label". That's how the field is called. The point is that is only available if the device is using the version 2 of LUKS (LUKS2 for the friends). Likely LUKS3 (if it comes at some point) will support labels too.